crates/kitsune2/src/lib.rs

```diff
@@ -0,0 +1,318 @@
+        let kitsune_builder_1 = Arc::new(kitsune_builder_1);
+        kitsune_builder_1
+            .config
+            .set_module_config(&Tx5TransportModConfig {
+                tx5_transport: Tx5TransportConfig {
+                    server_url: server_url.clone().to_string(),
+                    signal_allow_plain_text: true,
+                },
+            })
+            .unwrap();
+        kitsune_builder_1
+            .config
+            .set_module_config(&K2GossipModConfig {
+                k2_gossip: K2GossipConfig {
+                    initiate_interval_ms: 100,
+                    min_initiate_interval_ms: 100,
+                    ..Default::default()
+                },
+            })
+            .unwrap();
+
+        let kitsune_handler_1 = Arc::new(HolochainKitsuneHandler {
+            url: std::sync::Mutex::new(
+                Url::from_str("ws://placeholder.url:1").unwrap(),
+            ),
+            notify_received_receiver: tokio::sync::Mutex::new(None),
+        });
+        let kitsune_1 = kitsune_builder_1
+            .kitsune
+            .create(kitsune_builder_1.clone(), kitsune_handler_1.clone())
+            .await
+            .unwrap();
+        let space_1 = kitsune_1.space(TEST_SPACE_ID).await.unwrap();
+
+        let kitsune_builder_2 =
+            Arc::new(default_builder().with_default_config().unwrap());
+        kitsune_builder_2
+            .config
+            .set_module_config(&Tx5TransportModConfig {
+                tx5_transport: Tx5TransportConfig {
+                    server_url: server_url.to_string(),
+                    signal_allow_plain_text: true,
+                },
+            })
+            .unwrap();
+        kitsune_builder_2
+            .config
+            .set_module_config(&K2GossipModConfig {
+                k2_gossip: K2GossipConfig {
+                    initiate_interval_ms: 100,
+                    min_initiate_interval_ms: 100,
+                    ..Default::default()
+                },
+            })
+            .unwrap();
+
+        // Create 2 agents.
+        let kitsune_handler_2 = Arc::new(HolochainKitsuneHandler {
+            url: std::sync::Mutex::new(
+                Url::from_str("ws://placeholder.url:2").unwrap(),
+            ),
+            notify_received_receiver: tokio::sync::Mutex::new(None),
+        });
+        let kitsune_2 = kitsune_builder_2
+            .kitsune
+            .create(kitsune_builder_2.clone(), kitsune_handler_2.clone())
+            .await
+            .unwrap();
+        let space_2 = kitsune_2.space(TEST_SPACE_ID).await.unwrap();
+
+        let agent_url_1 = kitsune_handler_1.url.lock().unwrap().clone();
+        let local_agent_1 = Arc::new(TestLocalAgent::default());
+        let agent_1 = AgentBuilder {
+            url: Some(Some(agent_url_1.clone())),
+            storage_arc: Some(DhtArc::FULL),
+            ..Default::default()
+        }
+        .build(local_agent_1.clone() as DynLocalAgent);
```

Comment:
As this is an integration test, should we rely on the bootstrap server module rather than doing this manually? (and the peer store injection below) ?